plugins/modules/aci_l3out_logical_interface_profile_ospf_policy.py

```diff
@@ -0,0 +1,333 @@
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        l3out=dict(type="str", aliases=["l3out_name"]),
+        node_profile=dict(type="str", aliases=["node_profile_name", "logical_node"]),
+        interface_profile=dict(type="str", aliases=["interface_profile_name", "logical_interface"]),
+        ospf_policy=dict(type="str", aliases=["name", "ospf_policy_name"]),
+        ospf_auth_type=dict(type="str", default=None, choices=[None, "default", "simple", "md5"]),
+        ospf_auth_key=dict(type="str", default="", no_log=True),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "l3out", "node_profile", "interface_profile", "ospf_policy"]],
+            ["state", "present", ["tenant", "l3out", "node_profile", "interface_profile", "ospf_policy"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    l3out = module.params.get("l3out")
+    node_profile = module.params.get("node_profile")
+    interface_profile = module.params.get("interface_profile")
+    ospf_policy = module.params.get("ospf_policy")
+    ospf_auth_type = module.params.get("ospf_auth_type")
+    ospf_auth_key = module.params.get("ospf_auth_key")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="l3extOut",
+            aci_rn="out-{0}".format(l3out),
+            module_object=l3out,
+            target_filter={"name": l3out},
+        ),
+        subclass_2=dict(
+            aci_class="l3extLNodeP",
+            aci_rn="lnodep-{0}".format(node_profile),
+            module_object=node_profile,
+            target_filter={"name": node_profile},
+        ),
+        subclass_3=dict(
+            aci_class="l3extLIfP",
+            aci_rn="lifp-[{0}]".format(interface_profile),
+            module_object=interface_profile,
+            target_filter={"name": interface_profile},
+        ),
+        subclass_4=dict(
+            aci_class="ospfIfP",
+            aci_rn="ospfIfP",
+            module_object=interface_profile,
+            target_filter={"name": interface_profile},
+        ),
+        child_classes=["ospfRsIfPol"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = [
+            dict(ospfRsIfPol=dict(attributes=dict(
+                tnOspfIfPolName=ospf_policy
+            )))
+        ]
+        if ospf_auth_key != "":
```

Comment:
would something like this make it shorter?

example assumes the key check for None but if valid reason for that "" it could be checking on that also I would say

```
config = dict(authType=ospf_auth_type)
if ospf_auth_key is not None:
    config.update(authKey=ospf_auth_key)

aci.payload(
        aci_class="ospfIfP",
        class_config=config,
        child_configs=child_configs
)
```